### ISSUE TITLE
Play/Pause delay

### ISSUE THREAD

**Issue by Akhilesh-T** [stax76/mpv.net#468]
There is a 1-4 seconds delay when I **play/pause** videos. However, **next frame/previous frame** shortcuts work fine.

![Screenshot_24072022-111751](https://user-images.githubusercontent.com/80357467/180634251-535a31c2-875e-4932-bf50-c5cd961837a1.png)

**Comment by stax76** [comment]
There is now a new issue template for questions too, because often insufficient info is provided:

Some type of questions like questions about a problem require filling out an issue template:

**Describe the problem**
A clear and concise description of what the problem is.

**To Reproduce**
To reproduce, try the last beta version using the CLI option `--config=no`.

In case of an audio or video problem, try to reproduce the problem using mpv instead of mpv.net.
If it can be reproduced with mpv, use the [mpv issue tracker](https://github.com/mpv-player/mpv/issues).

Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.
1. mpv.net version
2. Windows version
3. GPU name
4. Media info

**Comment by Akhilesh-T** [comment]
**Describe the problem**
- When I click the pause shortcuts (_spacebar_ or _right mouse button_), the video doesn't pause instantly, but takes at least 1 to 4 seconds to pause.

**To Reproduce**
**To reproduce, try the last beta version using the CLI option --config=no.**
- Sorry, I'm using the latest beta version on Windows 10, but haven't got a clue on how to use the CLI option. I tried this using cmd, but it gave me this error `'--config' is not recognized as an internal or external command, operable program or batch file.
`

**In case of an audio or video problem, try to reproduce the problem using mpv instead of mpv.net.**
- I tried this, but the issue wasn't reproducible on mpv

**Steps to reproduce the behavior:**

- Open a video
- Let it play for a few seconds
- Click spacebar

**Expected behavior**
The video should pause immediately.

**Additional context**
- The pausing delay occurs only after I let the video play for a few seconds
- The issue occurs with every video - local as well as streaming

---
- **mpv.net version**: v6.0.3.0
- **Windows version**: Windows 10.0.19044 Build 19044 (21H2)
- **GPU name**: Intel(R) HD Graphics
- **Media info**:

General
Complete name                            : C:\Users\Akhilesh\Desktop\Youtube\Relaxing Hiking Video🍁- Into the Wild _ 4K _ 🎧 On-720p.mp4
Format                                   : MPEG-4
Format profile                           : Base Media
Codec ID                                 : isom (isom/iso2/avc1/mp41)
File size                                : 122 MiB
Duration                                 : 14 min 22 s
Overall bit rate                         : 1 191 kb/s
Movie name                               : Relaxing Hiking Video🍁- Into the Wild | 4K | 🎧 On
Performer                                : Ronnie & Barty
Recorded date                            : 20211113
Writing application                      : Lavf59.8.100
Comment                                  : https://www.youtube.com/watch?v=23rEJ7-hYy0

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : Main@L3.1
Format settings                          : CABAC / 3 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 3 frames
Codec ID                                 : avc1
Codec ID/Info                            : Advanced Video Coding
Duration                                 : 14 min 22 s
Bit rate                                 : 1 057 kb/s
Width                                    : 1 280 pixels
Height                                   : 720 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 24.000 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.048
Stream size                              : 109 MiB (89%)
Title                                    : ISO Media file produced by Google Inc. Created on: 11/14/2021.
Writing library                          : x264 core 155 r2901 7d0ff22
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
Menus                                    : 3
Codec configuration box                  : avcC

Audio
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 14 min 22 s
Bit rate mode                            : Constant
Bit rate                                 : 128 kb/s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 44.1 kHz
Frame rate                               : 43.066 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 13.2 MiB (11%)
Title                                    : ISO Media file produced by Google Inc. Created on: 11/14/2021.
Default                                  : Yes
Alternate group                          : 1
Menus                                    : 3

Menu #1
ID                                       : 3
Format                                   : Timed Text
Codec ID                                 : text
Duration                                 : 14 min 22 s
Language                                 : English
Menu For                                 : 1,2
00:00:00.000                             : Let's start hiking :)
00:04:13.000                             : SkillShare Integration
00:06:13.000                             : Back to the wilderness
00:08:20.000                             : Campsite Day 1
00:09:24.000                             : The next morning
00:11:28.000                             : River Crossing
00:12:23.000                             : The final ascent

Menu #2
00:00:00.000                             : Let's start hiking :)
00:04:13.000                             : SkillShare Integration
00:06:13.000                             : Back to the wilderness
00:08:20.000                             : Campsite Day 1
00:09:24.000                             : The next morning
00:11:28.000                             : River Crossing
00:12:23.000                             : The final ascent

**Comment by stax76** [comment]
Instead of using the CLI option, you can also temporarily rename your mpv.conf file, so mpv cannot find it, this is done to ensure mpv using default values, to be sure the problem isn't caused by your settings. The file is located here:

C:\Users\username\AppData\Roaming\mpv.net\mpv.conf

It's likely a problem with the stream or with mpv, you can get better help here:

https://www.reddit.com/r/mpv

**Comment by Akhilesh-T** [comment]
Renaming _mpv.conf_ didn't work for me. I tried playing many other files. The issue still persists in mpv.net

I rechecked mpv player, but again, I didn't face this issue.

**Comment by Akhilesh-T** [comment]
After a lot of trial and error, I was able to fix this.

I replaced `Space      script-message-to mpvnet play-pause` with `Space          cycle pause` and that solved the problem

**Comment by stax76** [comment]
Still very mysterious, all it does is reading `playlist-count` before invoking `cycle pause`. There might be more issues having the same mysterious root cause.

```C#
        public static void PlayPause()
        {
            int count = Core.GetPropertyInt("playlist-count");

            if (count > 0)
                Core.Command("cycle pause");
            else if (App.Settings.RecentFiles.Count > 0)
            {
                ...
            }
        }
```
